receiver/datadogreceiver/internal/translator/sketches.go

```diff
@@ -0,0 +1,294 @@
+	// (Note: in Datadog's code, it is referred to as 'bias')
+	agentSketchOffset int32 = 1338
+
+	// The max limit for the index of a sketch bucket
+	// See https://github.com/DataDog/opentelemetry-mapping-go/blob/00c3f838161a00de395d7d0ed44d967ac71e43b9/pkg/quantile/ddsketch.go#L21
+	// and https://github.com/DataDog/opentelemetry-mapping-go/blob/00c3f838161a00de395d7d0ed44d967ac71e43b9/pkg/quantile/ddsketch.go#L127
+	maxIndex = math.MaxInt16
+)
+
+// Unmarshal the sketch payload, which contains the underlying Dogsketch structure used for the translation
+func (mt *MetricsTranslator) HandleSketchesPayload(req *http.Request) (sp []gogen.SketchPayload_Sketch, err error) {
+	buf := GetBuffer()
+	defer PutBuffer(buf)
+	if _, err := io.Copy(buf, req.Body); err != nil {
+		return sp, err
+	}
+
+	pl := new(gogen.SketchPayload)
+	if err := pl.Unmarshal(buf.Bytes()); err != nil {
+		return sp, err
+	}
+
+	return pl.GetSketches(), nil
+}
+
+func (mt *MetricsTranslator) TranslateSketches(sketches []gogen.SketchPayload_Sketch) pmetric.Metrics {
+	bt := newBatcher()
+	bt.Metrics = pmetric.NewMetrics()
+
+	for _, sketch := range sketches {
+		dimensions := parseSeriesProperties(sketch.Metric, "sketch", sketch.Tags, sketch.Host, mt.buildInfo.Version, mt.stringPool)
+		metric, metricID := bt.Lookup(dimensions)
+		metric.ExponentialHistogram().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+		dps := metric.ExponentialHistogram().DataPoints()
+
+		dps.EnsureCapacity(len(sketch.Dogsketches))
+
+		// The dogsketches field of the payload contains the sketch data
+		for i := range sketch.Dogsketches {
+			dp := dps.AppendEmpty()
+
+			err := sketchToDatapoint(sketch.Dogsketches[i], dp, dimensions.dpAttrs)
+			if err != nil {
+				// If a sketch is invalid, remove this datapoint
+				metric.ExponentialHistogram().DataPoints().RemoveIf(func(dp pmetric.ExponentialHistogramDataPoint) bool {
+					if dp.Positive().BucketCounts().Len() == 0 && dp.Negative().BucketCounts().Len() == 0 {
+						return true
+					}
+					return false
+				})
+				continue
+			}
+			stream := identity.OfStream(metricID, dp)
+			if ts, ok := mt.streamHasTimestamp(stream); ok {
+				dp.SetStartTimestamp(ts)
+			}
+			mt.updateLastTsForStream(stream, dp.Timestamp())
+		}
+	}
+
+	return bt.Metrics
+}
+
+func sketchToDatapoint(sketch gogen.SketchPayload_Sketch_Dogsketch, dp pmetric.ExponentialHistogramDataPoint, attributes pcommon.Map) error {
+	dp.SetTimestamp(pcommon.Timestamp(sketch.Ts * time.Second.Nanoseconds())) // OTel uses nanoseconds, while Datadog uses seconds
+
+	dp.SetCount(uint64(sketch.Cnt))
+	dp.SetSum(sketch.Sum)
+	dp.SetMin(sketch.Min)
+	dp.SetMax(sketch.Max)
+	dp.SetScale(scale)
+	dp.SetZeroThreshold(math.Exp(float64(1-agentSketchOffset) / (1 / gamma))) // See https://github.com/DataDog/sketches-go/blob/7546f8f95179bb41d334d35faa281bfe97812a86/ddsketch/mapping/logarithmic_mapping.go#L48
```

Comment:
Sorry, I was making the right calculation but commenting wrong . Also see line 266 down below and [playground](https://go.dev/play/p/TDTM2EQGjw0).
```suggestion
	dp.SetZeroThreshold(math.Exp(float64(1-agentSketchOffset) / (1 / math.Log(gamma)))) // See https://github.com/DataDog/sketches-go/blob/7546f8f95179bb41d334d35faa281bfe97812a86/ddsketch/mapping/logarithmic_mapping.go#L48
```